apps/src/templates/instructions/TopInstructionsCSP.jsx

```diff
@@ -155,31 +164,52 @@ class TopInstructionsCSP extends Component {
     // adjusts max height.
     this.props.setInstructionsRenderedHeight(Math.min(maxNeededHeight, 300));
 
+    const promises = [];
+
     if (this.props.viewAs === ViewType.Student) {
-      $.ajax({
-        url:
-          '/api/v1/teacher_feedbacks/get_feedbacks?student_id=' +
-          this.props.user +
-          '&level_id=' +
-          this.props.serverLevelId,
-        method: 'GET',
-        contentType: 'application/json;charset=UTF-8'
-      }).done(data => {
-        this.setState({feedbacks: data}, this.forceTabResizeToMaxHeight);
-      });
+      promises.push(
+        $.ajax({
+          url: `/api/v1/teacher_feedbacks/get_feedbacks?student_id=${user}&level_id=${serverLevelId}`,
+          method: 'GET',
+          contentType: 'application/json;charset=UTF-8'
+        }).done(data => {
+          this.setState({feedbacks: data});
+        })
+      );
     }
     //While this is behind an experiment flag we will only pull the rubric
     //if the experiment is enable. This should prevent us from showing the
     //rubric if not in the experiment.
     if (experiments.isEnabled(experiments.MINI_RUBRIC_2019)) {
-      $.ajax({
-        url: `/levels/${this.props.serverLevelId}/get_rubric/`,
-        method: 'GET',
-        contentType: 'application/json;charset=UTF-8'
-      }).done(data => {
-        this.setState({rubric: data}, this.forceTabResizeToMaxHeight);
-      });
+      promises.push(
+        $.ajax({
+          url: `/levels/${serverLevelId}/get_rubric/`,
+          method: 'GET',
+          contentType: 'application/json;charset=UTF-8'
+        }).done(data => {
+          this.setState({rubric: data});
+        })
+      );
     }
+
+    if (this.state.teacherViewingStudentWork) {
+      promises.push(
+        $.ajax({
+          url: `/api/v1/teacher_feedbacks/get_feedback_from_teacher?student_id=${studentId}&level_id=${serverLevelId}&teacher_id=${user}`,
```

Comment:
nice lift! it is always great to see these network requests getting moved farther up the rendering chain. and since moving them all the way up and out of react sounds like a headache, consolidating them into this one location seems like an excellent alternative.